apps/web/app/(authenticated)/(app)/app/keys/[keyId]/settings/update-key-expiration.tsx

```diff
@@ -10,12 +9,34 @@ import {
   CardHeader,
   CardTitle,
 } from "@/components/ui/card";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
 import { Input } from "@/components/ui/input";
-import { Label } from "@/components/ui/label";
 import { Switch } from "@/components/ui/switch";
 import { toast } from "@/components/ui/toaster";
+import { trpc } from "@/lib/trpc/client";
 import { cn } from "@/lib/utils";
-import { updateExpiration } from "./actions";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { format } from "date-fns";
+
+import { useRouter } from "next/navigation";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+const currentTime = new Date();
+const oneMinute = currentTime.setMinutes(currentTime.getMinutes() + 0.5);
```

Comment:
lol what's the story behind this?

Reply:
Taken from createkey. If you set min time and then take to long to click the save button it sets a value in the past and breaks. No one should really want to use a date and time that close but people do strange things.